src/generators.rs

```diff
@@ -983,6 +983,139 @@ pub fn directed_grid_graph(
     })
 }
 
+/// Generate an undirected heavy square graph.
```

Comment:
I think it would benefit the reader a lot to inclue (1) a reference to the paper (e.g., Fig 6 of https://arxiv.org/pdf/1907.09528.pdf ) and (2) an ASCII diagram. Here's my attempt at artistry:

```
...       S   ...
   \     / \
... D   D   D ...
    |   |   |
... F-S-F-S-F-...
    |   |   |
... D   D   D ...
    |   |   |
... F-S-F-S-F-...
    |   |   |
    .........
    |   |   |
... D   D   D ...
     \ /     \
...   S       ...
```

Reply:
There is a variant of the heavy square lattice (see Fig 10.b, right-hand side) whose edge case handling (lines 1054-1075 in the present PR) is a bit different. No comment on whether it's worth including as an option, but it is definitely worth mentioning in the docstring.

Reply:
Sure, that makes sense. I'll include both.